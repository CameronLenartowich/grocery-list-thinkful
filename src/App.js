import React from 'react';
import { Route, Link } from 'react-router-dom';
import './App.css';
import * as firebase from 'firebase';
import RoomList from './components/RoomList';
import ItemList from './components/ItemList';
import User from './components/Users';
import './style/app.css'

  // Set the configuration for your app
  const firebaseConfig = {
    apiKey: "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
    authDomain: "grocery-list-1ed7b.firebaseapp.com",
    databaseURL: "https://grocery-list-1ed7b.firebaseio.com",
    projectId: "grocery-list-1ed7b",
    storageBucket: "grocery-list-1ed7b.appspot.com",
    messagingSenderId: "833209511385",
    appId: "1:833209511385:web:31f159d5098c2a96"
  };
  firebase.initializeApp(firebaseConfig);

class App extends React.Component {
  constructor(props){
    super(props)

    this.state = {
      currentRoom: '',
      currentRoomKey: '',
      user: ''
    }
  }

  selectedRoom(value) {
    this.setState({ currentRoom: value.name });
    this.setState({ currentRoomKey: value.key })  
  }

  setUser(name) {
    if (name == null){
      this.setState({ user: '' })
    } else {
      this.setState({ user: name.displayName });
    }  
  }

  render() {
    return (
      <div className="App">
          <header>
            
            <div className="header row">
                <div className="userController col-md-5 col-6">
                  <User 
                    firebase={firebase}
                    setUser={this.setUser.bind(this)}
                    user={this.state.user}
                  />
                </div>

                <div className="titleContainer col-md-7 col-6">
                  <h1 className="title">Grocery List</h1>
                </div>
            </div>
            
          </header>
          
          <main>
            <div className="main row">
              <div className="RoomList col-md-3 col-sm-4 col-5">
                <RoomList
                  firebase={firebase}
                  callbackFromParent={this.selectedRoom.bind(this)}
                />
              </div>           
              <div className="ItemList col-md-9 col-sm-8 col-7">
                <ItemList 
                    firebase={firebase}
                    currentRoom={this.state.currentRoom}
                    currentRoomKey={this.state.currentRoomKey}
                    user={this.state.user}
                />
              </div>
            </div>
          </main>
      </div>
    );
  }
}

export default App;
