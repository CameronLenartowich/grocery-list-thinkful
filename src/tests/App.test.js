import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { exportAllDeclaration } from '@babel/types';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('App component', () => {
    it('renders without crashing', () => {
      const component = Enzyme.shallow(<App />);
       expect(component).toMatchSnapshot();
    });
});