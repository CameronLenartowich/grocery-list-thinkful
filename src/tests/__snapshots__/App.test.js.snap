// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Grocery List
    </h1>
    <User
      firebase={
        Object {
          "INTERNAL": Object {
            "ErrorFactory": [Function],
            "createFirebaseNamespace": [Function],
            "createSubscribe": [Function],
            "deepExtend": [Function],
            "extendNamespace": [Function],
            "factories": Object {
              "auth": [Function],
              "database": [Function],
              "firestore": [Function],
              "functions": [Function],
            },
            "node": Object {
              "XMLHttpRequest": [Function],
              "localStorage": Storage {},
              "sessionStorage": Storage {},
            },
            "registerService": [Function],
            "removeApp": [Function],
            "useAsService": [Function],
          },
          "SDK_VERSION": "6.0.2",
          "User": [Function],
          "__esModule": true,
          "app": [Function],
          "apps": Array [
            FirebaseAppImpl {
              "INTERNAL": Object {
                "addAuthTokenListener": [Function],
                "getToken": [Function],
                "getUid": [Function],
                "removeAuthTokenListener": [Function],
              },
              "automaticDataCollectionEnabled_": false,
              "firebase_": [Circular],
              "isDeleted_": false,
              "name_": "[DEFAULT]",
              "options_": Object {
                "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                "appId": "1:833209511385:web:31f159d5098c2a96",
                "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                "databaseURL": "https://grocery-list-1ed7b.firebaseio.com",
                "messagingSenderId": "833209511385",
                "projectId": "grocery-list-1ed7b",
                "storageBucket": "grocery-list-1ed7b.appspot.com",
              },
              "services_": Object {
                "auth": Object {
                  "[DEFAULT]": Object {
                    "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                    "appName": "[DEFAULT]",
                    "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                    "currentUser": null,
                  },
                },
              },
            },
          ],
          "auth": [Function],
          "database": [Function],
          "default": [Circular],
          "firestore": [Function],
          "functions": [Function],
          "initializeApp": [Function],
        }
      }
      setUser={[Function]}
      user=""
    />
  </header>
  <hr />
  <main>
    <RoomList
      callbackFromParent={[Function]}
      firebase={
        Object {
          "INTERNAL": Object {
            "ErrorFactory": [Function],
            "createFirebaseNamespace": [Function],
            "createSubscribe": [Function],
            "deepExtend": [Function],
            "extendNamespace": [Function],
            "factories": Object {
              "auth": [Function],
              "database": [Function],
              "firestore": [Function],
              "functions": [Function],
            },
            "node": Object {
              "XMLHttpRequest": [Function],
              "localStorage": Storage {},
              "sessionStorage": Storage {},
            },
            "registerService": [Function],
            "removeApp": [Function],
            "useAsService": [Function],
          },
          "SDK_VERSION": "6.0.2",
          "User": [Function],
          "__esModule": true,
          "app": [Function],
          "apps": Array [
            FirebaseAppImpl {
              "INTERNAL": Object {
                "addAuthTokenListener": [Function],
                "getToken": [Function],
                "getUid": [Function],
                "removeAuthTokenListener": [Function],
              },
              "automaticDataCollectionEnabled_": false,
              "firebase_": [Circular],
              "isDeleted_": false,
              "name_": "[DEFAULT]",
              "options_": Object {
                "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                "appId": "1:833209511385:web:31f159d5098c2a96",
                "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                "databaseURL": "https://grocery-list-1ed7b.firebaseio.com",
                "messagingSenderId": "833209511385",
                "projectId": "grocery-list-1ed7b",
                "storageBucket": "grocery-list-1ed7b.appspot.com",
              },
              "services_": Object {
                "auth": Object {
                  "[DEFAULT]": Object {
                    "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                    "appName": "[DEFAULT]",
                    "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                    "currentUser": null,
                  },
                },
              },
            },
          ],
          "auth": [Function],
          "database": [Function],
          "default": [Circular],
          "firestore": [Function],
          "functions": [Function],
          "initializeApp": [Function],
        }
      }
    />
    <ItemList
      currentRoom=""
      currentRoomKey=""
      firebase={
        Object {
          "INTERNAL": Object {
            "ErrorFactory": [Function],
            "createFirebaseNamespace": [Function],
            "createSubscribe": [Function],
            "deepExtend": [Function],
            "extendNamespace": [Function],
            "factories": Object {
              "auth": [Function],
              "database": [Function],
              "firestore": [Function],
              "functions": [Function],
            },
            "node": Object {
              "XMLHttpRequest": [Function],
              "localStorage": Storage {},
              "sessionStorage": Storage {},
            },
            "registerService": [Function],
            "removeApp": [Function],
            "useAsService": [Function],
          },
          "SDK_VERSION": "6.0.2",
          "User": [Function],
          "__esModule": true,
          "app": [Function],
          "apps": Array [
            FirebaseAppImpl {
              "INTERNAL": Object {
                "addAuthTokenListener": [Function],
                "getToken": [Function],
                "getUid": [Function],
                "removeAuthTokenListener": [Function],
              },
              "automaticDataCollectionEnabled_": false,
              "firebase_": [Circular],
              "isDeleted_": false,
              "name_": "[DEFAULT]",
              "options_": Object {
                "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                "appId": "1:833209511385:web:31f159d5098c2a96",
                "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                "databaseURL": "https://grocery-list-1ed7b.firebaseio.com",
                "messagingSenderId": "833209511385",
                "projectId": "grocery-list-1ed7b",
                "storageBucket": "grocery-list-1ed7b.appspot.com",
              },
              "services_": Object {
                "auth": Object {
                  "[DEFAULT]": Object {
                    "apiKey": "AIzaSyDiMKeTwz7LgFfpPRNVc3dDQYH_A4e48Uo",
                    "appName": "[DEFAULT]",
                    "authDomain": "grocery-list-1ed7b.firebaseapp.com",
                    "currentUser": null,
                  },
                },
              },
            },
          ],
          "auth": [Function],
          "database": [Function],
          "default": [Circular],
          "firestore": [Function],
          "functions": [Function],
          "initializeApp": [Function],
        }
      }
      user=""
    />
  </main>
</div>
`;
